### PR TITLE
detect-flowbits: add details for flowbits v6

### PR DESCRIPTION
**Pull request by hadiqaalamdar**
Task #6309

- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/6309) ticket: https://redmine.openinfosecfoundation.org/issues/6309
Previous PR: https://github.com/OISF/suricata/pull/9971

Describe changes:
- added the recommended changes from the last PR.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1526

### CODE REVIEW

**Comment by inashivb** [comment]
> The pgsql failures are due to a recent new Suri merge.

And that tells that you should rebase your s-v PR w latest master and resubmit @hadiqaalamdar

**Comment by hadiqaalamdar** [comment]
> And that tells that you should rebase your s-v PR w latest master and resubmit @hadiqaalamdar Edit: Make both suricata and s-v PRs current if they aren't

should I create the new PR now after rebasing or should I wait for further instructions on how to deal with `noalert`? Also, I'd like some feedback on my or operator implementation.

**Review by jufajardini — CHANGES_REQUESTED** [review]
Hi there! Left a comment about the `or` operand. Shivani may have more to say about it.

If you know how to advance with this part, maybe you can submit a new PR with everything rebased, and that part update, while we understand how to approach the `noalert` case. :)

File: src/detect-engine-analyzer.c [hunk 1]
```diff
@@ -861,6 +863,47 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
...
+                    case DETECT_FLOWBITS_CMD_TOGGLE:
+                        jb_set_string(js, "cmd", "toggle");
+                        break;
+                }
+                if (cd->cmd != DETECT_FLOWBITS_CMD_NOALERT) {
+                    jb_open_array(js, "names");
+                    if (cd->or_list_size == 0) {
+                        jb_append_string(js, VarNameStoreSetupLookup(cd->idx, VAR_TYPE_FLOW_BIT));
+                    } else if (cd->or_list_size > 0) {
+                        jb_set_string(js, "operator", "or");
```
[the remark this answers is not in the document]
Comment (hadiqaalamdar) [reply src/detect-engine-analyzer.c:895]: got it, I'll try this out. Thanks!

**Comment by hadiqaalamdar** [comment]
New PR: https://github.com/OISF/suricata/pull/10018

**Pull request closed without merging**